### PR TITLE
feat: create ConfirmTitle component

### LINKED ISSUE #21282: Create a component to show title and description of confirmation pages
We need to add a component to display title and description for the confirmation. The title and description content will depend of the type of the confirmation request.

Designs: https://www.figma.com/file/uACzbBjwU2tV30UuewYiTx/Confirmation-redesign-V2?type=design&node-id=775-28145&mode=design&t=fcwGIVZS3lkt9rOO-0

See Architecture Proposal for more info: https://www.notion.so/Confirmation-UI-UX-Redesign-c99de275e5914a27a5a578faf9aa6069#de28d103ea3e443f88fbae916ffba548

The component can be placed in /ui/components/app/confirm

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMWdkQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--5ffb82714644a7ce9d0fabcbce33fe7f64c95165/Screenshot%202023-10-10%20at%204.02.02%E2%80%AFPM.png)

### PR DESCRIPTION
**Pull request by digiwand**
## **Description**

New component to display the confirmation title and subtitle

Notes:
- storybook page created
- I put the custom hooks directly in the confirm/title directory because it is not being used outside of this directory

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/21282
Blocked By: https://github.com/MetaMask/metamask-extension/pull/21970

## **Screenshots/Recordings**

see blue box in below image:
confirmtitle

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by jpuri — DISMISSED** [review]
👍

**Review by jpuri — DISMISSED** [review]

File: ui/components/app/confirm/title/title.tsx
Comment (digiwand) [reply ui/components/app/confirm/title/title.tsx:19]: essentially, would we be using 

current:
```
<ConfirmTitle> 
```
or

passed values:
```
<ConfirmTitle title={} description={}>
```
?

File: app/_locales/en/messages.json
Comment (georgewrmarshall) [inline app/_locales/en/messages.json:796]: I believe single line sentences should not contain a period. cc @coreyjanssen https://www.notion.so/MetaMask-Content-Design-Style-Guide-7d0b7579786540afbe02780ebb203254?pvs=4#7ffcee3343bc4115991c7a7e0d2f1c71

Screenshot 2024-01-09 at 4 41 22 PM

Comment (digiwand) [reply app/_locales/en/messages.json]: lgtm! updated d6f448e191195ee96d29050f45c7c8e9272d55c4

thanks @georgewrmarshall

**Review by georgewrmarshall — DISMISSED** [review]
Content LGTM!

**Review by blackdevelopa — APPROVED** [review]

**Review by blackdevelopa — DISMISSED** [review]

**Review by georgewrmarshall — DISMISSED** [review]

File: ui/components/app/confirm/title/title.tsx
Comment (segun) [reply ui/components/app/confirm/title/title.tsx:19]: I think @jpuri mentioned we want all the components to handle their logic and not pass props.

**Review by segun — CHANGES_REQUESTED** [review]
Would be nice to add storybook as well.

**Comment by digiwand** [comment]
hi @segun, thanks for the review!

I was holding off on creating storybook pages for this reason:

> no storybook page created for this. We can display this within the confirm page storybook page(s) when we create them. see: https://storybook.js.org/docs/writing-stories/build-pages-with-storybook#args-composition-for-presentational-screens

however, I've added it now upon request since it could be useful. It's a simple storybook page for now, so it doesn't have controls yet we don't support other variants at this time

updated [5983f88](https://github.com/MetaMask/metamask-extension/pull/22270/commits/5983f8836aca54dff199439fe828de86e164a467)

**Comment by georgewrmarshall** [comment]
> hi @segun, thanks for the review!
> 
> I was holding off on creating storybook pages for this reason:
> 
> > no storybook page created for this. We can display this within the confirm page storybook page(s) when we create them. see: https://storybook.js.org/docs/writing-stories/build-pages-with-storybook#args-composition-for-presentational-screens
> 
> however, I've added it now upon request since it could be useful. It's a simple storybook page for now, so it doesn't have controls yet we don't support other variants at this time
> 
> updated [5983f88](https://github.com/MetaMask/metamask-extension/pull/22270/commits/5983f8836aca54dff199439fe828de86e164a467)

I also think a storybook file no matter how simple the component is can be helpful :)

**Review by georgewrmarshall — DISMISSED** [review]

File: ui/components/app/confirm/title/title.tsx
Comment (digiwand) [reply ui/components/app/confirm/title/title.tsx:19]: great mention @segun. I also thought this, though, for our ConfirmInfo component, @jpuri requested to pass props.

double-checking here

**Comment by digiwand** [comment]
gotchya, thanks for the direction on storybook pages here @segun & @georgewrmarshall!

File: ui/components/app/confirm/title/title.tsx
Comment (jpuri) [reply ui/components/app/confirm/title/title.tsx:19]: Hey @digiwand : sorry for delayed response, this code looks good. I think we do not need to pass values.

**Review by jpuri — APPROVED** [review]

**Review by georgewrmarshall — APPROVED** [review]

**Comment by digiwand** [comment]
Screenshot 2024-01-29 at 4 41 00 PM

hi @segun,
not seeing a way to dismiss or mark the change request as complete. may I get a rereview please?

**Review by segun — APPROVED** [review]

**Pull request merged**
